cmd/tink-worker/cmd/root.go

```diff
@@ -0,0 +1,173 @@
+				return err
+			}
+			return applyViper(viper, cmd)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			retryInterval, _ := cmd.Flags().GetDuration("retry-interval")
+			retries, _ := cmd.Flags().GetInt("retries")
+			// TODO(displague) is log-level no longer useful?
+			// logLevel, _ := cmd.Flags().GetString("log-level")
+			workerID, _ := cmd.Flags().GetString("id")
+			maxFileSize, _ := cmd.Flags().GetInt64("max-file-size")
+			timeOut, _ := cmd.Flags().GetDuration("timeout")
+			user, _ := cmd.Flags().GetString("registry-username")
+			pwd, _ := cmd.Flags().GetString("registry-password")
+			registry, _ := cmd.Flags().GetString("docker-registry")
+
+			logger.With("version", version).Info("starting")
+			if setupErr := client.Setup(); setupErr != nil {
+				return setupErr
+			}
+
+			ctx := context.Background()
+			if timeOut > 0 {
+				var cancel context.CancelFunc
+				ctx, cancel = context.WithTimeout(ctx, timeOut)
+				defer cancel()
+			}
+
+			conn, err := tryClientConnection(logger, retryInterval, retries)
+			if err != nil {
+				return err
+			}
+			rClient := pb.NewWorkflowSvcClient(conn)
+
+			regConn := internal.NewRegistryConnDetails(registry, user, pwd, logger)
+			worker := internal.NewWorker(rClient, regConn, logger, registry, retries, retryInterval, maxFileSize)
+
+			err = worker.ProcessWorkflowActions(ctx, workerID)
+			if err != nil {
+				return errors.Wrap(err, "worker Finished with error")
+			}
+			return nil
+		},
+	}
+
+	rootCmd.Flags().Duration("retry-interval", retryIntervalDefault, "Retry interval in seconds")
+
+	rootCmd.Flags().Duration("timeout", time.Duration(defaultTimeoutMinutes*time.Minute), "Max duration to wait for worker to complete")
+
+	rootCmd.Flags().Int("max-retry", retryCountDefault, "Maximum number of retries to attempt")
+
+	rootCmd.Flags().Int64("max-file-size", defaultMaxFileSize, "Maximum file size in bytes")
+
+	// rootCmd.Flags().String("log-level", "info", "Sets the worker log level (panic, fatal, error, warn, info, debug, trace)")
+
+	rootCmd.Flags().StringP("id", "i", "", "Sets the worker id")
+	must(rootCmd.MarkFlagRequired("id"))
+
+	rootCmd.Flags().StringP("docker-registry", "r", "", "Sets the Docker registry")
+	must(rootCmd.MarkFlagRequired("docker-registry"))
+
+	rootCmd.Flags().StringP("registry-username", "u", "", "Sets the registry username")
+	must(rootCmd.MarkFlagRequired("registry-username"))
+
+	rootCmd.Flags().StringP("registry-password", "p", "", "Sets the registry-password")
+	must(rootCmd.MarkFlagRequired("registry-password"))
+
+	return rootCmd
+}
+
+// createViper creates a Viper object configured to read in configuration files
+// (from various paths with content type specific filename extensions) and load
+// environment variables that start with TINK_WORKER.
+func createViper() (*viper.Viper, error) {
+	v := viper.New()
+	v.AutomaticEnv()
+	v.SetConfigName("tink-worker")
+	v.AddConfigPath("/etc/tinkerbell")
+	v.AddConfigPath(".")
```

Comment:
did we lose somewhere in $HOME?

Reply:
I could add `$HOME/.config/something/` back, I thought it might be easier to figure that out later. (the preference for a $HOME subpath and a filename in that path may complicate this).

Reply:
makes sense